clap_blocks/src/compactor.rs

```diff
@@ -27,15 +27,52 @@ pub struct CompactorConfig {
     )]
     pub write_buffer_partition_range_end: i32,
 
-    /// Percentage of least recent data we want to split to reduce compacting non-overlapped data
-    /// Must be between 0 and 100. Default is 100, which won't split the resulting file.
+    /// Max number of level-0 files we want to compact with level-1 each time
+    /// Default: 3
+    /// TODO: verify the theory that if the streaming works  as expected,
+    /// we do not need to limit this number
+    #[clap(
+        long = "--compaction-max-number-level-0-files",
+        env = "INFLUXDB_IOX_COMPACTION_MAX_NUMBER_LEVEL_0_FILES",
+        default_value = "3",
+        action
+    )]
+    pub compaction_max_number_level_0_files: i32,
+
+    /// Desired max size that a compacted result won't be split and written into one file
+    /// It is a `desired` value because we decide whether to split or not based on
+    /// the data before compaction.
+    /// Default is 30,000,000 (30MB)
```

Comment:
```suggestion
    /// Desired max size of compacted parquet files 
    /// It is a target desired value rather than a guarantee.
    /// Default is 30,000,000 (30MB)
```

I am not sure if this rephrasing is correct; I didn't quite understand the  "won't be split and written into one file" (because if it is split, then wouldn't it be in *more* than one file?)